### PR TITLE
Set primary key of implicit_action table to action_id

### LINKED ISSUE #31140: Application Table implicit_action doesn't have primary key
### Describe the bug

There is no primary key for the table implicit_action ... For example if an application database is running MYSQL as master-replica setup it needs a primary key for the insert script

### To Reproduce

Check the implicit_action table

### Expected behavior

There is a primary key for that table

### Logs

_No response_

### Information about your Metabase installation

```JSON
1.46.4 and master
```

### Severity

blocking some users to upgrade due to their MYSQL setup

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by calherries**
Resolves https://github.com/metabase/metabase/issues/31140

This PR sets the PK of the `implicit_action` table to the `action_id` column.

Previously the `implicit_action` table had no primary key.

### CODE REVIEW

File: resources/migrations/000_migrations.yaml [hunk 1]
```diff
@@ -14812,6 +14812,47 @@ databaseChangeLog:
...
+      changes:
+        - addPrimaryKey:
+            tableName: implicit_action
+            columnNames: action_id
+            constraintName: pk_implicit_action
+
+  - changeSet:
+      id: v47.00-037
+      author: calherries
+      comment: Added 0.47.0 - Add foreign key constraint on implicit_action.action_id
```
Comment (qnkhuat) [inline resources/migrations/000_migrations.yaml:14845]: do you think we need an index?

**Review by qnkhuat — APPROVED** [review]
I think we'd want an index for implict_action.action_id

File: resources/migrations/000_migrations.yaml (hunk 1 above)
Comment (calherries) [reply resources/migrations/000_migrations.yaml:14845]: Indices are automatically created for primary keys in postgres, mysql, and h2.
Postgres: https://www.postgresql.org/docs/current/indexes-unique.html#:~:text=PostgreSQL%20automatically%20creates%20a%20unique,mechanism%20that%20enforces%20the%20constraint.
MySQL: https://www.tutorialspoint.com/is-the-primary-key-automatically-indexed-in-mysql#:~:text=Yes%2C%20primary%20key%20is%20automatically,primary%20key%20to%20be%20indexed.
H2: https://groups.google.com/g/h2-database/c/Hi_JnOPM9vI?pli=1

Comment (qnkhuat) [reply resources/migrations/000_migrations.yaml:14845]: this is a FK right? FKs are not automatically indexed https://github.com/metabase/metabase/issues/30924

Comment (calherries) [reply resources/migrations/000_migrations.yaml:14845]: It's a PK. See the migration above https://github.com/metabase/metabase/pull/31292/files#diff-241462177925c37373615bb7d440f151756dc4e629f4fbd5a6ac26bf6a688d4fR14835

**Pull request merged**